Code/Framework/AzCore/AzCore/Math/Uuid.inl

```diff
@@ -17,19 +17,37 @@
 
 namespace AZ::UuidInternal
 {
+    // Lookup table to convert ascii values for 0-9, a-f, and A-F to hex values in the range 0-15.
+    // Lambda expression populates the CharToHexDigit lookup table at compile time(it can't be invoked at runtime)
+    // It is invoked immediately
+    inline constexpr auto CharToHexDigit = []() constexpr
+    {
+        AZStd::array<AZStd::byte, 256> charToHexTable{};
+        AZStd::fill(charToHexTable.begin(), charToHexTable.end(), AZStd::numeric_limits<AZStd::byte>::max());
```

Comment:
Defining a constant for this "invalid" Uuid character value (i.e. `AZ::UuidInternal::InvalidValue`) would be helpful.

Something else that I noticed, though...  
Do we actually have `numeric_limits` defined for `AZStd::byte`?  On a whim I put in this line of code in a file and it says it's failing:
`static_assert(AZStd::numeric_limits<AZStd::byte>::max() == AZStd::byte(255), "Why is byte max not 255?");`

Anyone else seeing that?

Reply:
For clarity, this static assert passes:
`static_assert(AZStd::numeric_limits<AZStd::byte>::max() == AZStd::byte(0), "Why is byte max 0?");`

Reply:
We probably just need to add specializations for `AZStd::byte`: https://en.cppreference.com/w/cpp/types/numeric_limits

Reply:
AZStd::byte is std::byte [under the hood](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/base.h#L34) which is an enum class.

Reply:
Instead of using numeric_limits for the std::byte enum we can just use AZStd::byte(255) instead

Reply:
That works too, then I'd definitely prefer to define this as a constant in `Uuid` somewhere.  Fortunately, the old `GetValue` code appears to be the only place where numeric_limits<byte> was being used.  Still, people may walk into this trap in the future, thinking it's defined when it isn't.

Reply:
Updated to use Uuid::Internal::InvalidValue and AZStd::byte(255) instead of AZStd::numeric_limits<AZStd::byte>::max()